inference/usage_tracking/persistent_queue.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import sqlite3
+import time
+
+from typing_extensions import Any, Dict, List
+
+from inference.core.env import MODEL_CACHE_DIR
+from inference.core.logger import logger
+
+
+class PersistentQueue:
+    def __init__(self, db_file_path: str = os.path.join(MODEL_CACHE_DIR, "usage.db")):
```

Comment:
maybe if we have cls method init that would initialise everything and pass connection to ready DB we could:
* avoid potential exceptions from constructor - which is just my personal preference
* have an easy way to mock connections in tests

Reply:
to mock `db_file_path` is set to ":memory:", I added some tests covering this class, please have a look.

Also, instantiate this class in UsageCollector with catch-them-all exception handling so UsageCollector falls back to regular Queue - if `os.path.join` fails it's caught in upper layer and graceful fall-back is applied